lib/RoleDetails/utils.js

```diff
@@ -100,3 +100,35 @@ export async function createUserRolesRequests(previousSelectedUsers, currentSele
 
   return requests;
 }
+
+/**
+ * Asynchronously checks the authentication of Keycloak users based on the provided user IDs.
+ *
+ * @param {Object} options - The options object.
+ * @param {Array} options.userIds - An array of user IDs to check authentication for.
+ * @param {Object} options.queryClient - The query client object for fetching data.
+ * @param {Object} options.ky - The ky instance for making HTTP requests.
+ */
+
+export async function checkAndGetNoneKeycloakAuthUsers({ roleId, userIds, queryClient, ky }) {
+  const keycloakNoneExistingUsers = [];
+
+  const checkByUserId = async (userId) => {
+    try {
+      await queryClient.fetchQuery(
+        `jit-auth-role-${roleId}-user-${userId}`,
+        () => ky.get(`users-keycloak/auth-users/${userId}`)
+      );
+    } catch (error) {
+      if (error?.response?.status === 404) {
```

Comment:
I don't love that a normal use-case path needs to go through try/catch. `react-query` has a more imperative way of handling non-200 response codes. Can you give this a try? https://stackoverflow.com/a/71537971/330204

Reply:
I don't love that either, and handling business logic in a catch block has always seemed unfair to frontend devs to me. 
But since we need to make requests for every userId in the list from the 'find-user Pluggable' on save button click, we have to use [queryClient.fetchQuery](https://tanstack.com/query/latest/docs/reference/QueryClient#queryclientfetchquery) which unfortunately doesn't provide the thing like queryClient.fetchQuery( , {onError})